### PR TITLE
MultiRewards Tests v1

### PR DESCRIPTION
**Pull request by zcor**
Created initial batch of tests for MultiRewards contract

### CODE REVIEW

**Comment by zcor** [comment]
Thanks!

File: tests/test_withdraw.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# ERC20 function should fail on non-owner
+def test_recover_erc20_onlyowner(multi_reward, reward_token, accounts, alice, bob, chain):
+    multi_reward.stake(10**18, {'from': alice })
+    reward_token.approve(multi_reward, 10**19)
+    multi_reward.setRewardsDistributor(reward_token, alice, {'from': alice})
+    multi_reward.notifyRewardAmount(reward_token, 10**10, {'from': alice})
+    _init_amount = reward_token.balanceOf(bob)
+    chain.sleep(60)
+    chain.mine()
+    assert multi_reward.earned(alice, reward_token)  > 0
```
Comment (iamdefinitelyahuman) [inline tests/test_withdraw.py:83]: there are 4 tests opening with this exact logic sequence..  they could be placed into a different module, and this logic moved to a module-scoped "setup" fixture

File: tests/test_withdraw.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+# Can user retrieve reward after a time cycle?
+def test_get_reward(multi_reward, reward_token, accounts, alice, chain):
+    multi_reward.stake(10**18, {'from': alice })
+    reward_token.approve(multi_reward, 10**19)
+    multi_reward.setRewardsDistributor(reward_token, alice, {'from': alice})
+    multi_reward.notifyRewardAmount(reward_token, 10**10, {'from': alice})
+    _init_amount = reward_token.balanceOf(alice)
+    chain.sleep(60)
+    chain.mine()
```
Comment (iamdefinitelyahuman) [inline tests/test_withdraw.py:40]: could also be expressed as `chain.mine(timedelta=60)`

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+
+import pytest
+from brownie import CurveTokenV1, CurveTokenV2, CurveTokenV3
+from brownie.test import given, strategy
+from hypothesis import settings
+
+pytest_plugins = [
+    "fixtures.accounts",
```
Comment (iamdefinitelyahuman) [inline tests/conftest.py:9]: Separating fixtures across multiple modules is useful for large test suites, but for a project of this size I feel it's overkill.  I would keep all fixtures within the base `conftest.py`

File: contracts/tokens/CurveTokenV1.vy [hunk 4]
```diff
@@ -0,0 +1,171 @@
+# @version 0.1.0b16
```
Comment (iamdefinitelyahuman) [inline contracts/tokens/CurveTokenV1.vy:1]: Prefer the use of [`brownie-token-tester`](https://github.com/iamdefinitelyahuman/brownie-token-tester) over including token contracts for testing.  It keeps the repo a bit cleaner.

File: tests/test_stake.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# Can other users stake?
+@pytest.mark.parametrize("idx", range(1,5))
+def test_place_stake(multi_reward, accounts, base_token, reward_token, idx):
+    base_token.approve(multi_reward, 1000000, {'from': accounts[idx]})
+    multi_reward.stake(1000000, {'from': accounts[idx] })
+    assert multi_reward.balanceOf(accounts[idx]) > 0
+
+# Unbanked users should not be able to stake
+@pytest.mark.parametrize("idx", range(6,10))
+def test_place_stake(multi_reward, accounts, base_token, reward_token, idx):
```
Comment (iamdefinitelyahuman) [inline tests/test_stake.py:27]: duplicate test function names here, only one of these will run

File: tests/test_approve.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+# Can contract be paused and unpaused
+def test_contract_pausable(multi_reward, accounts, alice):
+    multi_reward.setPaused(True, {'from': alice})
+    multi_reward.setPaused(False, {'from': alice})
+    assert multi_reward.paused() is False
+
+# Can contract be paused by a rando?
+@pytest.mark.parametrize('idx', range(1,10))
+def test_contract_pausable(multi_reward, accounts, alice, idx):
```
Comment (iamdefinitelyahuman) [inline tests/test_approve.py:31]: duplicate use of `test_contract_pausable` for a function name

**Review by iamdefinitelyahuman — COMMENTED** [review]
Not a bad start!

Please also add [`black`](https://github.com/psf/black), [`flake8`](https://gitlab.com/pycqa/flake8) and [`isort`](https://github.com/timothycrosley/isort) to the repo for linting.

**Pull request closed without merging**